diff --git a/theories/Type/Loader.v b/theories/Type/Loader.v
index c3e7ed9..16cbcca 100644
--- a/theories/Type/Loader.v
+++ b/theories/Type/Loader.v
@@ -11,7 +11,7 @@
 Require Import Extraction.
 
 (** This exports tactics *)
-Declare ML Module "coq-equations.plugin".
+Declare ML Module "equations_plugin:coq-equations.plugin".
 
 Set Warnings "-notation-overridden".
 From Equations Require Export Init Signature.
